### DISCUSSION TITLE
Can I  assign the tasks for specifc work when the fastapi startup?

### DISCUSSION THREAD

**Discussion by kennard520** [fastapi/fastapi#12126] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
No
```

### Description

In My project, I will listene the multiple topic,  If I start listener is lifespan method, It will excute by every work. So, I want to share the listener to works. How to do that?

### Operating System

Linux

### FastAPI Version

0.112.2

### Pydantic Version

2.8.2

### Python Version

Python 3.12.2

**Comment by YuriiMotov** [comment, marked as answer]
Use the distributed Lock mechanism.
Similar question with answer: https://github.com/fastapi/fastapi/discussions/12793
